add docs for gathering septop and abfe (#1737)
OpenFreeEnergy/openfe: docs/reference/cli/gather.rst

``gather`` command
====================

Currently, ``openfe gather`` is only able to gather results from Relative Binding Free Energy (RBFE) calculations.

> To gather results from ABFE or SepTop protocols, you may use the experimental :ref:`openfe gather-abfe <gather-abfe>` and  :ref:`openfe gather-septop <gather-septop>` CLI commands, but please note that these commands are still under development and liable to change in future releases, and meant to be used only for exploratory work.

Reviewer: ```suggestion
To gather results from ABFE or SepTop protocols, use the experimental :ref:`openfe gather-abfe <gather-abfe>` and  :ref:`openfe gather-septop <gather-septop>` CLI commands.

.. warn:: These commands are still under development and liable to change in future releases, and meant to be used only for exploratory work.
```

Okay not sure if the formatting is right on this BUT I think using a warning callout will be more effective in communicating the warning
